### PR TITLE
[ticket/16790] Remove unused code

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-16790

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16790

### CODE REVIEW

**Comment by 3D-I** [comment]
Are the `*CP` modules still to be kept as they are or planned to be ported to controllers?

**Comment by rubencm** [comment]
They should be converted to controllers, but if I dont remember wrong mrgoldy had an issue doing it

File: phpBB/includes/acp/acp_board.php
Comment (CHItA) [inline phpBB/includes/acp/acp_board.php:480]: That's not in line with our coding guidelines.

**Comment by CHItA** [comment]
This is really hard to review and I'm not sure how much benefit it has, as most of these files should be rewritten anyway at some point. We also require `break`s pretty much everywhere, which is arguably silly, but that should be updated first, before this is merged.

IMO, these changes while probably do no harm, do not provide much benefit either. To be clear, the code you touched is pretty horrible, my problem is that it will still be horrible after we merge this. I don't think these changes address any of the real issues, or make the code good.

File: phpBB/includes/acp/acp_attachments.php
Comment (CHItA) [reply phpBB/includes/acp/acp_attachments.php:228]: Giving up any performance or extra memory so that my IDE doesn't color a variable grey doesn't seem like a sound tradeoff. Also, this exact pattern is in the PHP docs: https://www.php.net/manual/en/function.array-key-first.php#refsect1-function.array-key-first-notes

**Review by marc1706 — CHANGES_REQUESTED** [review]

File: tests/console/config/config_test.php
Comment (marc1706) [inline tests/console/config/config_test.php:17]: Class name shouldn't be changed. (we should probably move tests to namespaces at some point ;) )

**Review by marc1706 — CHANGES_REQUESTED** [review]

File: tests/console/config/config_test.php
Comment (rubencm) [reply tests/console/config/config_test.php:17]: I changed it to be able to run tests individually without having to run all test suite. File names in tests are not consistent with the class names.

**Comment by rubencm** [comment]
Dont merge, i am going to clean functions_compress and keep only the code that is used
A decent component for compress/decompress can be done later for the one click updater

**Review by marc1706 — APPROVED** [review]

**Pull request merged**
